src/libcore/ops/drop.rs

```diff
@@ -1,85 +1,136 @@
-/// Used to run some code when a value goes out of scope.
-/// This is sometimes called a 'destructor'.
+/// Custom code within the destructor.
 ///
-/// When a value goes out of scope, it will have its `drop` method called if
-/// its type implements `Drop`. Then, any fields the value contains will also
-/// be dropped recursively.
+/// When a value is no longer needed, Rust will run a "destructor" on that value.
```

Comment:
So here we call it "destructor" and later we call the same thing "drop glue"?

Reply:
"destructor" could be another term we could use for "drop glue", but I am not sure if that isn't confusing as people might also think it refers to `Drop::drop`. But in that case we should later explain which of the two concepts that we explain in more detail, is meant by "destructor" -- or probably we should never use that term again, and only use it here in the introduction before we established the terminology.

In fact, that made me think of a structure like that:

> When a value is no longer needed, Rust will run a "destructor" on that value.
>
>  [...]
>
> Destructors in Rust consist of two components:
> * Implementations of the `Drop` trait, that customize the destructor for values of a particular type.
> * The compiler-generated "drop glue", that automatically calls `Drop::drop` on a value and then recursively does the same for all its fields.

So to avoid the cyclic dependency, we basically introduce both terms at the same time. What do you think?

Reply:
yeah :sweat_smile:

I now rewrote the section mentioning "drop glue"

Reply:
Didn't see your second comment, seems like a good approach :+1: